components/system-broker/internal/director/client.go

```diff
@@ -50,47 +50,42 @@ type GraphQLizer interface {
 	PackageInstanceAuthRequestInputToGQL(in schema.PackageInstanceAuthRequestInput) (string, error)
 }
 
-func NewGraphQLClient(gqlClient Client, gqlizer GraphQLizer, gqlFieldsProvider GqlFieldsProvider) *GraphQLClient {
+func NewGraphQLClient(gqlClient Client, gqlizer GraphQLizer, gqlFieldsProvider GqlFieldsProvider, c *Config) *GraphQLClient {
 	return &GraphQLClient{
 		gcli: gqlClient,
 		//queryProvider:     queryProvider{}, - gqlizers are better
 		inputGraphqlizer:  gqlizer,
 		outputGraphqlizer: gqlFieldsProvider,
+		pageSize:          c.PageSize,
 	}
 }
 
 type GraphQLClient struct {
 	gcli              Client
 	inputGraphqlizer  GraphQLizer
 	outputGraphqlizer GqlFieldsProvider
+	pageSize          int
 	//queryProvider     queryProvider
 }
 
-func (c *GraphQLClient) FetchApplications(ctx context.Context) (*ApplicationsOutput, error) {
+func (c *GraphQLClient) FetchApplications(ctx context.Context) (ApplicationsOutput, error) {
 	response := ApplicationsOutput{}
 
 	query := fmt.Sprintf(`query {
-			result: applications {
+			result: applications(first: %%d, after: %%q) {
 					%s
 			}
 	}`, c.outputGraphqlizer.Page(c.outputGraphqlizer.ForApplication()))
+	queryGenerator := func(pageSize int, page string) string {
```

Comment:
this could then become something like variableResolver and accepts a map of variables to resolve with